<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Fits the parameters of a BiSSeCo model by maximum likelihood — fitbisseco • musseco</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Fits the parameters of a BiSSeCo model by maximum likelihood — fitbisseco"><meta name="description" content='This method estimates a pair of relative fitness coefficients from pathogen
phylogenies representing differences in transmissibility and differences in
within-host fitness.
Relative fitness is paramaterised by 1) the parameter "alpha" which represents
the fold-change in within-host replicative fitness and
2) the parameter "omega" which represents the fold-change in between-host
replicative fitness. See details for more information.'><meta property="og:description" content='This method estimates a pair of relative fitness coefficients from pathogen
phylogenies representing differences in transmissibility and differences in
within-host fitness.
Relative fitness is paramaterised by 1) the parameter "alpha" which represents
the fold-change in within-host replicative fitness and
2) the parameter "omega" which represents the fold-change in between-host
replicative fitness. See details for more information.'></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">musseco</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/musseco.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Functions</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/emvolz/musseco/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Fits the parameters of a BiSSeCo model by maximum likelihood</h1>
      <small class="dont-index">Source: <a href="https://github.com/emvolz/musseco/blob/HEAD/R/musseco1.R" class="external-link"><code>R/musseco1.R</code></a></small>
      <div class="d-none name"><code>fitbisseco.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This method estimates a pair of relative fitness coefficients from pathogen
phylogenies representing differences in transmissibility and differences in
within-host fitness.
Relative fitness is paramaterised by 1) the parameter "alpha" which represents
the fold-change in within-host replicative fitness and
2) the parameter "omega" which represents the fold-change in between-host
replicative fitness. See details for more information.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">fitbisseco</span><span class="op">(</span></span>
<span>  <span class="va">tr</span>,</span>
<span>  <span class="va">isvariant</span>,</span>
<span>  <span class="va">Tg</span>,</span>
<span>  <span class="va">mu</span>,</span>
<span>  Net <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  theta0 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">15</span>, omega <span class="op">=</span> <span class="fl">0.95</span>, yscale <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  augment_likelihood <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  optim_parms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  mlesky_parms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  res <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-tr">tr<a class="anchor" aria-label="anchor" href="#arg-tr"></a></dt>
<dd><p>an ape::phylo representing a time-scaled phylogeny. These
can be computed with the treedater package</p></dd>


<dt id="arg-isvariant">isvariant<a class="anchor" aria-label="anchor" href="#arg-isvariant"></a></dt>
<dd><p>vector of type boolean with length equal to ape::Ntip(tr).
Each element is TRUE if the corresponding element in tr$tip.label is a
variant type</p></dd>


<dt id="arg-tg">Tg<a class="anchor" aria-label="anchor" href="#arg-tg"></a></dt>
<dd><p>Generation time, i.e. the mean time elapsed between generations.
Units of this variable should match those used in tr$edge.length and 1/mu,
e.g. days or years</p></dd>


<dt id="arg-mu">mu<a class="anchor" aria-label="anchor" href="#arg-mu"></a></dt>
<dd><p>Molecular clock rate of evolution</p></dd>


<dt id="arg-net">Net<a class="anchor" aria-label="anchor" href="#arg-net"></a></dt>
<dd><p>NULL or a matrix with two columns giving the effective population size.
If NULL, the effective population size will be computed with the mlesky
package. The first column should be time since some point in the past, and
the second column should be an estimate of the effective population size.
Time zero should correspond to the root of the tree</p></dd>


<dt id="arg-theta-">theta0<a class="anchor" aria-label="anchor" href="#arg-theta-"></a></dt>
<dd><p>Initial guess of (log) parameter values: alpha, omega, and Ne scale.
Can be a named vector or in the aforementioned order.</p></dd>


<dt id="arg-augment-likelihood">augment_likelihood<a class="anchor" aria-label="anchor" href="#arg-augment-likelihood"></a></dt>
<dd><p>if TRUE (default), will combine the coalescent
likelihood with a binomial likelihood of sampling variants or ancestral
types under the assumption of random sampling and mutation-selection balance</p></dd>


<dt id="arg-optim-parms">optim_parms<a class="anchor" aria-label="anchor" href="#arg-optim-parms"></a></dt>
<dd><p>optional list of parameters passed to <a href="https://rdrr.io/r/stats/optim.html" class="external-link">optim</a>
when fitting the model</p></dd>


<dt id="arg-mlesky-parms">mlesky_parms<a class="anchor" aria-label="anchor" href="#arg-mlesky-parms"></a></dt>
<dd><p>optional list of parameters passed to <a href="https://rdrr.io/pkg/mlesky/man/mlskygrid.html" class="external-link">mlskygrid</a> if
estimating Ne(t)</p></dd>


<dt id="arg-res">res<a class="anchor" aria-label="anchor" href="#arg-res"></a></dt>
<dd><p>Integer number time steps used in coalescent likelihood</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional parameters are passed to <a href="http://emvolz-phylodynamics.github.io/phydynR/reference/colik.html" class="external-link">colik</a></p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A fitted BiSSeCo model with coef and summary methods</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Pathogen phylogenies are assumed to be reconstructed from population-based
random samples of pathogen genomes and at most one sequence per host.
Phylogenies should be time-scaled, and an estimate of the molecular clock rate
of evolution should be provided, such as estimated with the treedater
package.
If not provided, the effective population size over time is estimated with the
mlesky package, and this estimate is also provided in the returned BiSSeCo
fit.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># load tree simulated with TiPS</span></span></span>
<span class="r-in"><span><span class="va">tr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/tips_params_1_rep_2.rds"</span>, package <span class="op">=</span> <span class="st">"musseco"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># set to TRUE tips that are from variant</span></span></span>
<span class="r-in"><span><span class="va">isvariant</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span> <span class="va">tr</span><span class="op">$</span><span class="va">tip.label</span>, pattern <span class="op">=</span> <span class="st">'V_'</span> <span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">isvariant</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">isvariant</span>, <span class="va">tr</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#this will take a few minutes to run</span></span></span>
<span class="r-in"><span><span class="co">#the values of mu and Tg was based on true values used to simulate the trees</span></span></span>
<span class="r-in"><span><span class="co">#see vignette for more details.</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="va">fb_au</span> <span class="op">&lt;-</span> <span class="fu">fitbisseco</span><span class="op">(</span> <span class="va">tr</span>,</span></span>
<span class="r-in"><span>                     <span class="va">isvariant</span>,</span></span>
<span class="r-in"><span>                     Tg <span class="op">=</span> <span class="fl">10.2</span>,</span></span>
<span class="r-in"><span>                     mu <span class="op">=</span> <span class="fl">0.0016</span>,</span></span>
<span class="r-in"><span>                     Net <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>                     theta0 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">.75</span>,<span class="fl">1</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                     augment_likelihood <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                     optim_parms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                     mlesky_parms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>tau <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>                                         tau_lower <span class="op">=</span> <span class="fl">0.1</span>,</span></span>
<span class="r-in"><span>                                         tau_upper <span class="op">=</span> <span class="fl">1e7</span>,</span></span>
<span class="r-in"><span>                                         ncpu <span class="op">=</span> <span class="fl">1</span>, </span></span>
<span class="r-in"><span>                                         model <span class="op">=</span> <span class="fl">1</span> <span class="op">)</span> <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span><span class="co"># you can load the results from here</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/fit.rda"</span>, package <span class="op">=</span> <span class="st">"musseco"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">fb_au</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Binary state speciation and extinction coalescent fit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               2.5% 97.5%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> alpha  0.910 0.570 1.453</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> omega  0.918 0.874 0.964</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> yscale 0.844 0.769 0.926</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Likelihood: -5331.4195 </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Erik Volz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

