<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Estimating fitness using BiSSeCO • musseco</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Estimating fitness using BiSSeCO">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">musseco</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/musseco.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Functions</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/emvolz/musseco/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Estimating fitness using BiSSeCO</h1>
                        <h4 data-toc-skip class="author">Fabricia F. Nascimento</h4>
            
            <h4 data-toc-skip class="date">2025-02-12</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/emvolz/musseco/blob/HEAD/vignettes/musseco.Rmd" class="external-link"><code>vignettes/musseco.Rmd</code></a></small>
      <div class="d-none name"><code>musseco.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette will demonstrate how to to estimate within-host and between-host
transmission fitness using the package <code>musseco</code> and the BiSSeCo
(Binary-State Speciation and extinction Coalescent) model.</p>
</div>
<div class="section level2">
<h2 id="basic-requirements">Basic requirements<a class="anchor" aria-label="anchor" href="#basic-requirements"></a>
</h2>
<p>This vignette assumes that you have the following packages already installed:</p>
<ul>
<li>
<a href="https://github.com/emvolz/musseco" class="external-link">musseco</a>: implements the likelihood
function to estimate fitness.</li>
<li>
<a href="https://github.com/emmanuelparadis/ape" class="external-link">ape</a>: used for manipulation of
phylogenetic trees.</li>
<li>
<a href="https://cran.r-project.org/web/packages/TiPS/vignettes/TiPS.html" class="external-link">TiPS</a>: used
to simulate a phylogenetic tree according to an epidemiological model.</li>
<li>
<a href="https://ggplot2.tidyverse.org/" class="external-link">ggplot2</a>: used to visualize the results.</li>
</ul>
<div class="section level3">
<h3 id="load-the-necessary-packages">Load the necessary packages:<a class="anchor" aria-label="anchor" href="#load-the-necessary-packages"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://emvolz.github.io/musseco/">musseco</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://gitlab.in2p3.fr/ete/tips/" class="external-link">TiPS</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/emmanuelparadis/ape" class="external-link">ape</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="simulated-phylogenetic-tree">Simulated phylogenetic tree<a class="anchor" aria-label="anchor" href="#simulated-phylogenetic-tree"></a>
</h2>
<p>To illustrate how we can use the R package <code>musseco</code> to estimate relative fitness,
we will use a tree that was simulated with the R package <code>TiPS</code>.</p>
<p>This tree can be load as:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/tips_params_1_rep_2.rds"</span>, package <span class="op">=</span> <span class="st">"musseco"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="epidemiological-model">Epidemiological model<a class="anchor" aria-label="anchor" href="#epidemiological-model"></a>
</h3>
<p>This tree was simulate in a way that we know the true values of within-host and
between-host replicative fitness.</p>
<p>The epidemiological model we used are described by the following equations,
describing how the number of individuals carrying the ancestral (<span class="math inline">\(Y_a\)</span>) and variant
(<span class="math inline">\(Y_v\)</span>) sequences changes with time.</p>
<p><span class="math inline">\(\dot{Y_a} = \beta Y_a (1 - \dfrac{N}{K}) + \mu_{va}Y_v - \mu_{av}Y_a - \gamma Y_a\)</span></p>
<p><span class="math inline">\(\dot{Y_v} = \beta(1+s)Y_v (1 - \dfrac{N}{K}) - \mu_{va}Y_v + \mu_{av}Y_a - \gamma Y_v\)</span></p>
<p>where the rate of new infections was defined as logistic growth controlled by the
parameter <span class="math inline">\(\beta\)</span> for <span class="math inline">\(Y_a(t)\)</span> and <span class="math inline">\(\beta(1+s)\)</span>, where <span class="math inline">\(s\)</span> is the selection coefficient,
for <span class="math inline">\(Y_v(t)\)</span>.</p>
<p>The carry capacity was defined by the parameter <span class="math inline">\(K\)</span> and was fixed at
10,000 individuals.</p>
<p>The rate <span class="math inline">\(\mu_{av}\)</span> defined how individuals carrying the ancestral state <span class="math inline">\(Y_a(t)\)</span>
mutated to the <span class="math inline">\(Y_v(t)\)</span>.
The rate <span class="math inline">\(\mu_{va}\)</span> defined how individuals carrying the variant mutation
reversed to the ancestral state, and this is also the molecular clock rate of
evolution.</p>
<p>The rate <span class="math inline">\(\gamma\)</span> denoted the mortality rate.</p>
<p>In this vignette will not focus on how this tree was simulated,
but the scripts to simulate the tree can be found <a href="https://github.com/thednainus/musseco_simulations/blob/main/coalescent_sim/TiPS_simulations/simulate_TiPS_trees.R" class="external-link">here</a>.</p>
<p>Table 1 shows the parameter values used to simulate the tree</p>
<table class="table table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:table1">Table 1: </span>Parameter symbols and values
</caption>
<thead><tr>
<th style="text-align:left;">
Parameter
</th>
<th style="text-align:left;">
Symbol
</th>
<th style="text-align:left;">
Values
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Growth rate
</td>
<td style="text-align:left;">
<span class="math inline">\(\beta\)</span>
</td>
<td style="text-align:left;">
0.216
</td>
</tr>
<tr>
<td style="text-align:left;">
Carrying capacity
</td>
<td style="text-align:left;">
<span class="math inline">\(K\)</span>
</td>
<td style="text-align:left;">
10,000
</td>
</tr>
<tr>
<td style="text-align:left;">
Mortality rate
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma\)</span>
</td>
<td style="text-align:left;">
1/10.2
</td>
</tr>
<tr>
<td style="text-align:left;">
Transition rate from <span class="math inline">\(Y_v\)</span> to <span class="math inline">\(Y_a\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\mu_{va}\)</span>
</td>
<td style="text-align:left;">
0.0016
</td>
</tr>
<tr>
<td style="text-align:left;">
Transition rate from <span class="math inline">\(Y_a\)</span> to <span class="math inline">\(Y_v\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\mu_{av}\)</span>
</td>
<td style="text-align:left;">
0.00176
</td>
</tr>
<tr>
<td style="text-align:left;">
Selection coefficient
</td>
<td style="text-align:left;">
<span class="math inline">\(s\)</span>
</td>
<td style="text-align:left;">
-0.1
</td>
</tr>
<tr>
<td style="text-align:left;">
Between-host replicative fitness
</td>
<td style="text-align:left;">
<span class="math inline">\(\omega\)</span>
</td>
<td style="text-align:left;">
0.9
</td>
</tr>
<tr>
<td style="text-align:left;">
Within-host replicative fitness
</td>
<td style="text-align:left;">
<span class="math inline">\(\alpha\)</span>
</td>
<td style="text-align:left;">
1.1
</td>
</tr>
</tbody>
</table>
<p>Based on this parameter values, we know the true value of <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\omega\)</span>.</p>
<p>Note that <span class="math inline">\(\alpha = \mu_{av}/\mu_{va}\)</span> and <span class="math inline">\(\omega = 1 + s\)</span>.</p>
</div>
</div>
<div class="section level2">
<h2 id="estimation-of-parameter-values-using-fitbisseco">Estimation of parameter values using fitbisseco<a class="anchor" aria-label="anchor" href="#estimation-of-parameter-values-using-fitbisseco"></a>
</h2>
<p>To estimate the within-host and between-host replicative fitness using the
simulated phylogenetic tree, we will use the function <em>fitbisseco</em> from the
<code>musseco</code> package.</p>
<p>Note that we need to define which tips of the phylogenetic tree carries the
variant state.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">isvariant</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span> <span class="va">tr</span><span class="op">$</span><span class="va">tip.label</span>, pattern <span class="op">=</span> <span class="st">'V_'</span> <span class="op">)</span></span>
<span><span class="va">isvariant</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">isvariant</span>, <span class="va">tr</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span></span></code></pre></div>
<p>We also need to define the value of mu, which is the molecular clock
rate of evolution. In our example, we used the rate for HIV-1 subtype C
<span class="citation">(Patiño-Galindo JÁ 2017)</span> as we were modelling HIV-1 dynamics.</p>
<p>We also need to define the generation time (Tg).
The generation time can be defined as <span class="math inline">\(1/\gamma\)</span>.</p>
<p>Based on our Table 1, we know the values of mu and gamma.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># gamma value used in our epidemiological model</span></span>
<span><span class="co">#Tg = 1/gamma</span></span>
<span><span class="va">gamma</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">/</span><span class="fl">10.2</span></span>
<span><span class="co"># mu (molecular clock) value used in our epidemiological model</span></span>
<span><span class="va">mu</span> <span class="op">&lt;-</span> <span class="fl">0.0016</span></span></code></pre></div>
<p>Now we can run the <em>fitbisseco</em> function using the following command:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fb_au</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fitbisseco.html">fitbisseco</a></span><span class="op">(</span> <span class="va">tr</span>,</span>
<span>                     <span class="va">isvariant</span>,</span>
<span>                     Tg <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="va">gamma</span>,</span>
<span>                     mu <span class="op">=</span> <span class="va">mu</span>,</span>
<span>                     Net <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                     theta0 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">.75</span>,<span class="fl">1</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                     augment_likelihood <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     optim_parms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                     mlesky_parms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>tau <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                                         tau_lower <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>                                         tau_upper <span class="op">=</span> <span class="fl">1e7</span>,</span>
<span>                                         ncpu <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                                         model <span class="op">=</span> <span class="fl">1</span> <span class="op">)</span> <span class="op">)</span></span></code></pre></div>
<p>Note the following options:</p>
<ul>
<li>
<code>tr</code> is the <code><a href="https://rdrr.io/pkg/ape/man/read.tree.html" class="external-link">ape::phylo</a></code> object, and this must be a time-scaled phylogeny.</li>
<li>
<code>isvariant</code> vector of type boolean with lentgh equal to the number of tips in
the phylogenetic tree. Each element is <code>TRUE</code> if the corresponding element in tr$tip.label
is a variant type.</li>
<li>
<code>Tg</code> is the generation time, i.e. the mean time elapsed between generations.</li>
<li>
<code>mu</code> is the molecular clock rate of evolution;</li>
<li>
<code>Net</code> (default <code>NULL</code>): but user can also provide a matrix with two columns
giving the effective population size. If NULL, the effective population
size will be computed with the <a href="https://github.com/emvolz-phylodynamics/mlesky" class="external-link">mlesky</a>
R package. The first column should be time since some point in the past, and
the second column should be an estimate of the effective population size.
Time zero should correspond to the root of the tree.</li>
<li>
<code>theta0</code>: Initial guess of (log) parameter values: alpha, omega, and Ne scale.</li>
</ul>
<p>For details on the other parameters see <code><a href="../reference/fitbisseco.html">?musseco::fitbisseco</a></code>.</p>
<p>But user can also provide additional parameter values to use with optim when
fitting the model by maximum likelihood, mlskygrid function from the <a href="https://github.com/emvolz-phylodynamics/mlesky" class="external-link">mlesky</a>
R package and the <a href="https://emvolz-phylodynamics.github.io/phydynR/reference/colik.html" class="external-link">colik</a>
function from the <a href="https://emvolz-phylodynamics.github.io/phydynR/" class="external-link">phydynR</a> package.</p>
<p>In addition, the <em>fitbisseco</em> function implements two types of likelihoods:
the ‘coalescent’ and the ‘augmented’ likelihood.
Both methods are based on the conditional density of a genealogy given epidemic
and demographic parameters.
The coalescent likelihood (by setting augment_likelihood = FALSE) relies solely
on the phylogenetic tree to infer the fitness parameters.
The augmented likelihood combines in a simple sum, the coalescent likelihood with
a binomial likelihood of sampling variants or ancestral types under the assumption
of random sampling and mutation-selection balance.
The augmented likelihood is the default option in the <em>fitbisseco</em> function.</p>
<div class="section level3">
<h3 id="obtaining-the-estimated-parameter-values">Obtaining the estimated parameter values<a class="anchor" aria-label="anchor" href="#obtaining-the-estimated-parameter-values"></a>
</h3>
<p>The fitbisseco function will take a few minutes to run.
You can load the fit here</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/fit.rda"</span>, package <span class="op">=</span> <span class="st">"musseco"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fb_au</span></span>
<span><span class="co">#&gt; Binary state speciation and extinction coalescent fit</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;               2.5% 97.5%</span></span>
<span><span class="co">#&gt; alpha  0.910 0.570 1.453</span></span>
<span><span class="co">#&gt; omega  0.918 0.874 0.964</span></span>
<span><span class="co">#&gt; yscale 0.844 0.769 0.926</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Likelihood: -5331.4195</span></span></code></pre></div>
<p>We can now plot alpha and omega in respect to the true values listed in Table 1.
The true values was those used to simulate the phylogenetic tree.</p>
<p>First we need to get the confidence interval as data.frame.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># threshold for approximate CIs </span></span>
<span><span class="va">MAXCOEFVAR</span> <span class="op">&lt;-</span> <span class="fl">1.5</span></span>
<span></span>
<span><span class="co">#variable names  </span></span>
<span><span class="va">vnames</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'alpha'</span>, <span class="st">'omega'</span>, <span class="st">'yscale'</span><span class="op">)</span></span>
<span><span class="co">#the best values for the parameters found using optim</span></span>
<span><span class="va">best_values</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">fb_au</span><span class="op">)</span><span class="op">[</span><span class="va">vnames</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># create a dataframe with the confidence interval</span></span>
<span><span class="va">odf</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">best_values</span><span class="op">)</span></span>
<span><span class="va">odf</span><span class="op">$</span><span class="va">lower_bound</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">fb_au</span><span class="op">)</span><span class="op">[</span><span class="va">vnames</span><span class="op">]</span><span class="op">)</span><span class="op">-</span><span class="va">fb_au</span><span class="op">$</span><span class="va">err</span><span class="op">*</span><span class="fl">1.96</span> <span class="op">)</span></span>
<span><span class="va">odf</span><span class="op">$</span><span class="va">upper_bound</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">fb_au</span><span class="op">)</span><span class="op">[</span><span class="va">vnames</span><span class="op">]</span><span class="op">)</span><span class="op">+</span><span class="va">fb_au</span><span class="op">$</span><span class="va">err</span><span class="op">*</span><span class="fl">1.96</span> <span class="op">)</span></span>
<span><span class="va">odf</span><span class="op">$</span><span class="va">upper_bound</span><span class="op">[</span> <span class="va">fb_au</span><span class="op">$</span><span class="va">err</span> <span class="op">&gt;</span> <span class="va">MAXCOEFVAR</span> <span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">Inf</span></span>
<span><span class="va">odf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span> <span class="va">odf</span>, <span class="fl">3</span> <span class="op">)</span></span>
<span></span>
<span><span class="co">#add the true values to data.frame</span></span>
<span><span class="va">odf</span><span class="op">[</span><span class="st">"true_values"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1.1</span>, <span class="fl">0.9</span>, <span class="cn">NA</span><span class="op">)</span></span>
<span><span class="va">odf</span><span class="op">[</span><span class="st">"Variable"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>, <span class="st">"omega"</span>, <span class="cn">NA</span><span class="op">)</span></span>
<span><span class="va">odf</span><span class="op">$</span><span class="va">Variable</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">odf</span><span class="op">$</span><span class="va">Variable</span><span class="op">)</span></span></code></pre></div>
<p>Plot we can plot the values for alpha and omega using ggplot.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">odf</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,<span class="op">]</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Variable</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">best_values</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">true_values</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"blue"</span>, size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_linerange.html" class="external-link">geom_errorbar</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>ymax <span class="op">=</span> <span class="va">upper_bound</span>, ymin <span class="op">=</span> <span class="va">lower_bound</span>, </span>
<span>                    width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">Variable</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"True value"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Confidence Interval"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>, legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:figure1"></span>
<img src="musseco_files/figure-html/figure1-1.png" alt="Bar plots showing the confidence interval for estimates of alpha and omega. Blue dot is the true value"><p class="caption">
Figure 1: Bar plots showing the confidence interval for estimates of alpha and omega. Blue dot is the true value
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Patino-Galindo_2017" class="csl-entry">
Patiño-Galindo JÁ, González-Candelas F. 2017. <span>“The Substitution Rate of <span>HIV-1</span> Subtypes: A Genomic Approach.”</span> <em>Virus Evolution</em> 3 (2): vex029.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Erik Volz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
